routes/user.js

```diff
@@ -277,11 +342,29 @@ router.post(
         return res.status(404).json({ msg: 'User not found', toEmail });
       }
 
+      //Cannot invite yourself
+      if (user.email === toEmail)
+        return res
+          .status(400)
+          .json({ msg: 'Sorry but you cannot invite yourself', toEmail });
+
+      //If toEmail is already in current user's contacts
+      const alreadyFriends = user.contacts.find(
+        (contact) => contact.email.toString() === toEmail,
+      );
+
+      if (alreadyFriends) {
+        return res
+          .status(400)
+          .json({ msg: 'Email already in user contacts.', toEmail });
+      }
```

Comment:
If every invitation is created and then sent, maybe we should do all checks in the creation and then none in the send so that 1) we don't create but not send one and 2) we don't repeating any code by checking the same thing twice

Reply:
true!  I've just been trying to cover all situations in each route because if I'm testing them independently in Postman I don't want conditions to pass that should not.

Reply:
oh, that's a good point. Then maybe we extra all the validation out into one function that can be used in both places?